laser_encoders/test_models_initialization.py

```diff
@@ -0,0 +1,58 @@
+import os
+import tempfile
+
+import pytest
+
+from laser_encoders.download_models import LaserModelDownloader
+from laser_encoders.language_list import LASER2_LANGUAGE, LASER3_LANGUAGE
+from laser_encoders.laser_tokenizer import initialize_tokenizer
+from laser_encoders.models import initialize_encoder
+
+
+def test_validate_achnese_models_and_tokenize_laser3(lang="acehnese"):
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        print(f"Created temporary directory for {lang}", tmp_dir)
+
+        downloader = LaserModelDownloader(model_dir=tmp_dir)
+        downloader.download_laser3(lang)
+        encoder = initialize_encoder(lang, model_dir=tmp_dir)
+        tokenizer = initialize_tokenizer(lang, model_dir=tmp_dir)
+
+        # Test tokenization with a sample sentence
+        tokenized = tokenizer.tokenize("This is a sample sentence.")
+
+    print(f"{lang} model validated successfully")
+
+
+def test_validate_english_models_and_tokenize_laser2(lang="english"):
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        print(f"Created temporary directory for {lang}", tmp_dir)
+
+        downloader = LaserModelDownloader(model_dir=tmp_dir)
+        downloader.download_laser2()
+
+        encoder = initialize_encoder(lang, model_dir=tmp_dir)
+        tokenizer = initialize_tokenizer(lang, model_dir=tmp_dir)
+
+        # Test tokenization with a sample sentence
+        tokenized = tokenizer.tokenize("This is a sample sentence.")
+
+    print(f"{lang} model validated successfully")
+
+
+def test_validate_kashmiri_models_and_tokenize_laser3(lang="kas"):
```

Comment:
By the way, I am still getting the `unhashable type: 'list'`error when I try to initialize a pipeline for Kashmiri, instead of our informative ValueError.

I guess the `LaserEncoderPipeline` interface is going to be our recommended one, so we need to include it in our test and make sure it works as expected. 

```
>>> laser3 = LaserEncoderPipeline(lang="kas")
2023-11-09 03:12:27,216 | INFO | laser_encoders.download_models |  - laser2.spm already downloaded
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/private/home/daviddale/dev/LASER/laser_encoders/models.py", line 383, in __init__
    self.encoder = initialize_encoder(
  File "/private/home/daviddale/dev/LASER/laser_encoders/models.py", line 355, in initialize_encoder
    downloader.download_laser3(lang=lang, spm=spm)
  File "/private/home/daviddale/dev/LASER/laser_encoders/download_models.py", line 91, in download_laser3
    result = self.get_language_code(LASER3_LANGUAGE, lang)
  File "/private/home/daviddale/dev/LASER/laser_encoders/download_models.py", line 73, in get_language_code
    lang_3_4 = language_list[lang]
TypeError: unhashable type: 'list'
```

Reply:
@avidale for this I have made some changes which gives Value error [here](https://github.com/facebookresearch/LASER/pull/257/commits/99ebbfd24c68613d43b8d4a4db5ec6c97a47c38e) is this correct?